test/unit/modules/executeWithRetryTest.js

```diff
@@ -0,0 +1,49 @@
+import chai from 'chai';
+import sinon from 'sinon';
+import { executeWithRetry } from '../../../src/modules/executeWithRetry';
+
+const { expect } = chai;
+let { executeCallback, stopCallback: timeoutCallback } = [null, null];
+
+describe('executeWithRetry', () => {
+  beforeEach(() => {
+    executeCallback = sinon.stub();
+    timeoutCallback = sinon.spy();
+  });
+
+  it('should call timeoutCallback when callback never return true', (done) => {
```

Comment:
Tests are having valid checks over test doubles, but it feels hard to get a use case from these tests. Think about having additionally tests with more contex like:
```
it('should set name after 85 milliseconds', (done) => {
    const expectedNametoBeSetAt = 120
    const setTheNameAt = 85
    let name = null
    const setName = () => {
      setTimeout(() => {
        name = 'BigCommerce'
      }, setTheNameAt)
    }

    executeWithRetry(25, 100, setName, () => {});

    setTimeout(() => {
      expect(name).equals('BigCommerce');
      done();
    }, expectedNametoBeSetAt);
  });


  it('should fail if setting the name happens after the max time', (done) => {
    const expectedNametoBeSetAt = 120
    const setTheNameAt = 130
    let name = null
    const setName = () => {
      setTimeout(() => {
        name = 'BigCommerce'
      }, setTheNameAt)
    }

    executeWithRetry(25, 100, setName, () => {});

    setTimeout(() => {
      expect(name).equals(null);
      done();
    }, expectedNametoBeSetAt);
  });
```

Reply:
I added two use cases